src/irr/video/COpenGLSampler.h

```diff
@@ -0,0 +1,110 @@
+#ifndef __IRR_C_OPENGL_SAMPLER_H_INCLUDED__
+#define __IRR_C_OPENGL_SAMPLER_H_INCLUDED__
+
+#include "irr/video/IGPUSampler.h"
+#include "COpenGLExtensionHandler.h"
+#include <algorithm>
+
+namespace irr {
+namespace video
+{
+
+class COpenGLSampler : public IGPUSampler
+{
+    //! Get native wrap mode value
+    inline static GLenum getTextureWrapMode(uint8_t clamp)
+    {
+        using namespace asset;
+        GLenum mode = GL_REPEAT;
+        switch (clamp)
+        {
+        case ETC_REPEAT:
+            mode = GL_REPEAT;
+            break;
+        case ETC_CLAMP_TO_EDGE:
+            mode = GL_CLAMP_TO_EDGE;
+            break;
+        case ETC_CLAMP_TO_BORDER:
+            mode = GL_CLAMP_TO_BORDER;
+            break;
+        case ETC_MIRROR:
+            mode = GL_MIRRORED_REPEAT;
+            break;
+        case ETC_MIRROR_CLAMP_TO_EDGE:
+            if (COpenGLExtensionHandler::FeatureAvailable[COpenGLExtensionHandler::IRR_EXT_texture_mirror_clamp])
```

Comment:
i think these extensions are already core in OpenGL 4.3

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r328105442